wagtail/wagtail_hooks.py

```diff
@@ -551,3 +552,14 @@ def format_message(self, log_entry):
                 }
             except (KeyError, TypeError):
                 return _("Workflow cancelled")
+
+
+@hooks.register("before_serve_page", order=0)
+def serve_options_response(page: Page, request: HttpRequest, *args, **kwargs):
```

Comment:
Issue: It feels surprising that `OPTIONS` methods aren't handled by the page class at all. Could this be moved into `serve`?

Reply:
@RealOrange That one gets a little tricky with `RoutablePageMixin`, which overrides `serve()` entirely. Since 'before_serve_page' is called from the view, it would be simpler to do it this way.

Reply:
@RealOrangeOne Another option here would be to move the logic to separete Page methods, so that `RoutablePageMixin` can easily access / call those from it's implementation of `serve()`

Reply:
Overriding `serve` is not just a `RoutablePageMixin` thing - it's [a documented recipe](https://docs.wagtail.org/en/stable/reference/pages/model_recipes.html#overriding-the-serve-method), and we routinely tell people that you can put anything in there that you would put in a Django view function. Requiring developers to fulfil certain conditions (such as calling helper methods) in their `serve` methods would be a big step backwards.

Reply:
Good point @gasman. So, to best address all this feedback, I've added methods to the `Page` model to allow for overriding, and adjusted the hook to initiate the checking and serving of the OPTIONS response.